### PR TITLE
Fixed Beam.apply_load() wrong result for high order loads with ends

### LINKED ISSUE #15301: Problem with Beam module
I think there is a problem with the Beam module when using a force with an order bigger than 1.
This is an example where R should be 1/3 and remains constant when changing the length of the beam, but unfortunately it's not the case.
I believe it has to do with the formation of the load equation, where the higher order compensation isn't been taken into consideration.

image

### PR DESCRIPTION
**Pull request by moorepants**
#### References to other Issues or PRs

Fixes #15301

#### Brief description of what is fixed or changed

We were incorrectly calculating the subtraction load if the user
supplied an end distance for a distributed load of order 2 and higher.
To solve this a Taylor series is used to compute the proper subtraction
function. @faradai2008 figured out the exact form of the series and it
has now been implemented with a loop summation. Tests have been added
for basic cases and prior incorrect tests have been modified.

#### Release Notes

* physics.continuum_mechanics
  * Fixed wrong result with high order loads with end points.

### CODE REVIEW

**Comment by moorepants** [comment]
@faradai2008 Would you mind reviewing this and trying out the fix?

**Comment by moorepants** [comment]
@jashan498 Would you mind reviewing this PR?

**Review by faradai2008 — APPROVED** [review]

**Review by jashan498 — APPROVED** [review]

**Review by jashan498 — APPROVED** [review]
This PR looks good!

**Comment by smichr** [comment]
FYI, `filldedent` was made to handle cases like this:

```
msg = ("If 'end' is provided the 'order' of the load cannot "
             "be negative, i.e. 'end' is only valid for distributed "
             "loads.")
raise ValueError(msg)
```
like this
```
msg = filldedent('''
    If 'end' is provided the 'order' of the load cannot
    be negative, i.e. 'end' is only valid for distributed
    loads.''')
raise ValueError(msg)
```
It produces this when raised:
```
Traceback (most recent call last):
  File "<string>", line 5, in <module>
ValueError: 
If 'end' is provided the 'order' of the load cannot be negative, i.e.
'end' is only valid for distributed loads.
```

**Comment by smichr** [comment]
doctests need modification

**Comment by moorepants** [comment]
Ok, tests are passing now.

File: doc/src/modules/physics/continuum_mechanics/beam_problems.rst [hunk 1]
```diff
@@ -346,36 +346,36 @@ away from start.
     >>> b.reaction_loads
     {M: 157/2, V: -49/2}
     >>> b.load
-           -2         -1                                                                      
-    157⋅<x>     49⋅<x>          0            0          1             -1          0          1
-    ───────── - ──────── + 4⋅<x>  - 4⋅<x - 2>  + <x - 3>  + 12⋅<x - 4>   - <x - 6>  - <x - 6> 
-        2          2                                                                          
+           -2         -1
+    157⋅<x>     49⋅<x>          0            0          1             -1            0          1
+    ───────── - ──────── + 4⋅<x>  - 4⋅<x - 2>  + <x - 3>  + 12⋅<x - 4>   - 6⋅<x - 6>  - <x - 6>
```
Comment (moorepants) [inline doc/src/modules/physics/continuum_mechanics/beam_problems.rst]: Just noticed this is slightly incorrect. It should be a 3 and a 6 as the coefficient.

File: sympy/physics/continuum_mechanics/beam.py [hunk 2]
```diff
@@ -391,12 +392,18 @@ def apply_load(self, value, start, order, end=None):
...
+                msg = ("If 'end' is provided the 'order' of the load cannot "
+                       "be negative, i.e. 'end' is only valid for distributed "
+                       "loads.")
+                raise ValueError(msg)
+            # NOTE : A Taylor series can be used to define the summation of
+            # singularity functions that subtract from the load past the end
+            # point such that it evaluates to zero past 'end'.
+            f = value * x**order
+            for i in range(0, order + 1):
+                self._load -= (f.diff(x, i).subs(x, end) *
```
Comment (moorepants) [inline sympy/physics/continuum_mechanics/beam.py]: this is supposed to be `.subs(x, end - start)`.

**Comment by moorepants** [comment]
Once tests pass this should now be ready. I caught an error in my first implementation and wrote a test for that mistake.

**Comment by moorepants** [comment]
Will merge in 24hrs if no objections, or please go ahead an merge.

**Pull request merged**